services/QuillLMS/lib/tasks/update_optimal_value_for_student_problem_reports.rake

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+namespace :update_optimal_value_for_student_problem_reports do
+  desc 'update optimal value for student problem reports based off of feedback history'
+  task :run => :environment do
+    StudentProblemReport.all.each do |report|
+      is_optimal = FeedbackHistory.find(report.feedback_history_id)&.optimal
+      report.optimal if !is_optimal.nil?
```

Comment:
I think you might have accidentally removed the assignment?

Reply:
Also, purely style, but I `unless` here is easier to parse: `report.optimal = true unless is_optimal.nil?`